sklego/meta/thresholder.py

```diff
@@ -33,21 +38,35 @@ class Thresholder(BaseEstimator, ClassifierMixin):
         The classes labels.
     """
 
-    def __init__(self, model, threshold: float, refit=False):
+    _required_parameters = ["model", "threshold"]
+
+    def __init__(self, model, threshold: float, refit=False, check_input=False):
         self.model = model
         self.threshold = threshold
         self.refit = refit
+        self.check_input = check_input
+
+    def _handle_unfitted(self, X, y, sample_weight):
+        sample_weight_ = _check_sample_weight(sample_weight, X)
+
+        self.estimator_ = clone(self.model)
+        if "sample_weight" in signature(self.estimator_.fit).parameters:
+            self.estimator_.fit(X, y, sample_weight=sample_weight_)
+        else:
+            if sample_weight is not None:
+                logging.warning("Estimator ignores sample_weight.")
+            self.estimator_.fit(X, y)
+        return self
 
     def _handle_refit(self, X, y, sample_weight=None):
         """Only refit when we need to, unless `refit=True` is present."""
         if self.refit:
-            self.estimator_ = clone(self.model)
-            self.estimator_.fit(X, y, sample_weight=sample_weight)
+            self._handle_unfitted(X, y, sample_weight)
         else:
             try:
-                _ = self.estimator_.predict(X[:1])
+                check_is_fitted(self.estimator_)
```

Comment:
From the docs:

> If None, estimator is considered fitted if there exist an attribute that ends with a underscore and does not start with double underscore.

I think it's a good proxy. The reason for this is that for non-array data, `X is not subscriptable`